packages/read-model/src/Product/Action/ProductActionViewFactory.php

```diff
@@ -25,4 +25,18 @@ public function createFromProduct(Product $product, string $absoluteUrl): Produc
             $absoluteUrl
         );
     }
+
+    /**
+     * @param array $hit
+     * @return \Shopsys\ReadModelBundle\Product\Action\ProductActionView
+     */
+    public function createFromHit(array $hit): ProductActionView
```

Comment:
I think createFromArray will be more versatile and the reading proper indexes should be mover outside read-model package into framework